webkit2gtk: update to 2.42.5

chrysos349 opened the pull request:
[ci skip]

**NOTES**
- `2.42.5` (and `2.42.4` before `.5` was released) was tested as a part of `gnome-45.*`
- Required for `gnome-45.*` (#48762)
- (Cross)-built for all archs that are checked in ci

#### Testing the changes
- I tested the changes in this PR: **YES**

#### Local build testing
- I built this PR locally for my native architecture, (x86_64)
- I built this PR locally for these architectures (if supported. mark crossbuilds):
  - aarch64-musl x
  - aarch64 x
  - armv7l x
  - armv6l-musl x
  - x86_64-musl
  - i686

oreo639 commented on srcpkgs/webkit2gtk/template:
```diff
@@ -29,14 +29,15 @@ makedepends="at-spi2-core-devel libjpeg-turbo-devel libpng-devel
  libgcrypt-devel libnuspell-devel libpsl-devel $(vopt_if x11 libXt-devel)
- $(vopt_if wayland 'MesaLib-devel libxkbcommon-devel wayland-devel wayland-protocols')"
+ $(vopt_if wayland 'MesaLib-devel libxkbcommon-devel wayland-devel wayland-protocols')
+ libjxl-devel"
```
Use `-DUSE_JPEGXL=OFF` for now. (jpegxl requires all dependents to be rebuilt for every minor release atm and webkit2gtk is a large dependent)

oreo639 commented on srcpkgs/webkit2gtk/template:
```diff
@@ -29,14 +29,15 @@ makedepends="at-spi2-core-devel libjpeg-turbo-devel libpng-devel
- $(vopt_if wayland 'MesaLib-devel libxkbcommon-devel wayland-devel wayland-protocols')"
+ $(vopt_if wayland 'MesaLib-devel libxkbcommon-devel wayland-devel wayland-protocols')
+ libjxl-devel"
 depends="gst-plugins-good1"
```
Add `_depends` like in https://github.com/void-linux/void-packages/pull/47600/files

oreo639 reviewed (commented):
Also, don't remove the reproducible patch.

chrysos349 commented:
@oreo639

I changed the template according to your recommendations. Thanks!

> Also, don't remove the reproducible patch.

I removed the patch, because the old one was rejected. I got a new one from chimera cports.

Could you explain why the reproducibility is important? No other distro (except for void and chimera) requires this patch.

oreo639 commented:
> Could you explain why the reproducibility is important? No other distro (except for void and chimera) requires this patch.

Helps with debugging. (and also ccache)

Also, regarding the `--reduce-memory-overheads`, were you getting OOMs when compiling to i686?

chrysos349 commented:
> (and also ccache)

So that's the reason I didn't see any speedup, despite the ccache being turned on.

> Also, regarding the `--reduce-memory-overheads` patch, were you getting OOMs when compiling to i686?

No, I didn't. It's not llvm, lol.

I thought it wouldn't hurt, and others might need it.

oreo639 commented:
> I thought it wouldn't hurt, and others might need it.

Imo, it can be added to LDFLAGS if it is needed.

chrysos349 commented:
Might as well remove the patch. I didn't have any problems without it.
